### PR TITLE
Added infer profile from credential logic

### PR DESCRIPTION
**Pull request by sethduffin**
Review/merge after #207 has been merged into `main`

PT Story: https://www.pivotaltracker.com/story/show/183083150

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
#207 is merged. Go ahead and rebase this PR, and fix lint errors.

**Comment by sethduffin** [comment]
@dmitrizagidulin Ok, this PR is ready to be merged.

File: package.json [hunk 1]
```diff
@@ -35,10 +35,9 @@
     "@react-navigation/native": "^6.0.1",
     "@react-navigation/stack": "^6.0.1",
     "@reduxjs/toolkit": "^1.6.1",
-    "@types/bson": "^4.0.5",
     "@types/text-encoding": "^0.0.36",
     "big-integer": "^1.6.49",
-    "bson": "^4.4.1",
+    "bson": "4.4.1",
```
Comment (dmitrizagidulin) [inline package.json:40]: Why the pinning of the bson version here?

**Review by dmitrizagidulin — APPROVED** [review]
:+1:

File: package.json (hunk 1 above)
Comment (sethduffin) [reply package.json:40]: Realm has `bson@4.4.1` as a dependency and it was clashing with our unpinned version of `bson` causing a couple of typescript errors.

**Pull request merged**